### ISSUE TITLE
Add support for tree-based classifiers from the MLJ library

### ISSUE THREAD

**Issue by RaunoArike** [JuliaTrustworthyAI/CounterfactualExplanations.jl#197]
Since the Feature Tweak generator works on tree-based models, but the package only supports gradient-based models as of now, we will integrate the package with the MLJ library to add support for `DecisionTreeClassifier` and `RandomForestClassifier` from the MLJ library.

To do:

- [x] Implement a new abstract type for non-differentiable models
- [x] Implement a new base type for tree-based models
- [x] Enable the use of `DecisionTreeClassifier` and `RandomForestClassifier` as concrete tree-based models
- [x] Implement the generic methods `logits` and `probs` for tree-based models
- [x] Add documentation
- [x] Add tests
